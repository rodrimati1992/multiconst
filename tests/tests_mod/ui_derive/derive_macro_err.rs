use multiconst::FieldType;


#[derive(FieldType)]
struct Foo(#[field_type(crate = hello)] u32);


#[derive(FieldType)]
struct Bar(#[field_type(world)] u32);


#[derive(FieldType)]
#[field_type(aaaaa)]
struct Baz( u32);

#[derive(FieldType)]
#[field_type(aaaaa)]
struct Qux( u32);


#[derive(FieldType)]
#[field_type(pub fooasdasd)]
struct Another(u32);

// ensuring that errors are emitted for all attributes
#[derive(FieldType)]
#[field_type(foooo)]
#[field_type(bar)]
#[field_type(pub fooasdasd)]
struct ManyAttrs(u32);


fn main(){}