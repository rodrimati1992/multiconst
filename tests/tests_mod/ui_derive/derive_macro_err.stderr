error: expected `pub` or `priv`
 --> tests/tests_mod/ui_derive/derive_macro_err.rs:5:25
  |
5 | struct Foo(#[field_type(crate = hello)] u32);
  |                         ^^^^^

error: expected `pub` or `priv`
 --> tests/tests_mod/ui_derive/derive_macro_err.rs:9:25
  |
9 | struct Bar(#[field_type(world)] u32);
  |                         ^^^^^

error: expected one of: `crate`, `pub`, `priv`
  --> tests/tests_mod/ui_derive/derive_macro_err.rs:13:14
   |
13 | #[field_type(aaaaa)]
   |              ^^^^^

error: expected one of: `crate`, `pub`, `priv`
  --> tests/tests_mod/ui_derive/derive_macro_err.rs:17:14
   |
17 | #[field_type(aaaaa)]
   |              ^^^^^

error: unexpected token
  --> tests/tests_mod/ui_derive/derive_macro_err.rs:22:18
   |
22 | #[field_type(pub fooasdasd)]
   |                  ^^^^^^^^^

error: expected one of: `crate`, `pub`, `priv`
  --> tests/tests_mod/ui_derive/derive_macro_err.rs:27:14
   |
27 | #[field_type(foooo)]
   |              ^^^^^

error: expected one of: `crate`, `pub`, `priv`
  --> tests/tests_mod/ui_derive/derive_macro_err.rs:28:14
   |
28 | #[field_type(bar)]
   |              ^^^

error: unexpected token
  --> tests/tests_mod/ui_derive/derive_macro_err.rs:29:18
   |
29 | #[field_type(pub fooasdasd)]
   |                  ^^^^^^^^^
